src/main/java/com/google/devtools/build/lib/syntax/StringModule.java

```diff
@@ -278,23 +278,23 @@ public String strip(String self, Object charsOrNone) {
             type = String.class,
             doc = "The string to replace with."),
         @Param(
-            name = "maxsplit",
+            name = "count",
```

Comment:
I think renaming this would be a breaking change. May need to keep the old name for now.

Reply:
Yes, it's clearly breaking! The question is if it's a change that's allowed to be made according to the compatability policy, the [Backwards Compatability document](https://docs.bazel.build/versions/master/backward-compatibility.html) does not mention what's allowed and not in these edge cases.

To draw some inspiration from other projects: the [Go1 compatability document](https://golang.org/doc/go1compat) states:

>  If a compiler or library has a bug that violates the specification, a program that depends on the buggy behavior may break if the bug is fixed. We reserve the right to fix such bugs.

I think that this is an example of that, as the Starlark implementation in Bazel does not satisfy the [spec](https://github.com/bazelbuild/starlark/blob/master/spec.md#stringreplace).

I'll leave it up to you and the team to figure this out, but I do think that it would be nice to allow to make some breaking changes without increasing the major-version number in the future.

Reply:
This is definitely a breaking change which is not allowed by our policy -- apologies this is not clear by the policy doc, but this is definitely going to break some users and cause some pain if we change it.

We should avoid renaming this for now.
The parameter is tagged with `legacyNamed = true`, so, subject to the rollout of https://github.com/bazelbuild/bazel/issues/8147, this parameter will be positional-only in the future. We can then rename the parameter safely for documentation purposes.

Reply:
To clarify, there's two breaking changes: the rename and the behavior for negative values.

The rename is subsumed by #8147, so we should definitely not do any extra work here. You can add a comment reminding us to get back to it:

> // TODO(#8147): rename param to "count" once it's positional-only.

The behavioral change for negatives is also breaking. I feel like it's not very likely people will be broken by it, but it seems like the kind of thing we should guard with a flag. Look for other examples of using incompatible flags in Starlark builtins -- for instance, `incompatibleDisableDepsetItems` in the depset constructor later in this file, and its corresponding definitions in `StarlarkSemantics` and `StarlarkSemanticsOptions`.

Reply:
Ok, let's do this the proper way, by marking the negative (or zero/default) value behaviour change as breaking.

Other than adding the flag and the handling of it, is there anything else that needs to be done? What's the process for creating issues with the `incompatible-change` and `breaking-change-*` labels?

Reply:
When you file the GitHub issue, you can cc me so that I add the labels.
If your change is ready this week, we can include it in Bazel 3.2.

Thanks!

Reply:
@alandonovan points out to me that we actually already flipped *and deleted* `--incompatible_restrict_named_params`, so I'll do the rename as part of the follow-up.